src/compiler/types.ts

```diff
@@ -1493,6 +1499,8 @@ module ts {
     // Class and interface types (TypeFlags.Class and TypeFlags.Interface)
     export interface InterfaceType extends ObjectType {
         typeParameters: TypeParameter[];           // Type parameters (undefined if non-generic)
+        outerTypeParameters: TypeParameter[];      // Outer type parameters (undefined if none)
+        localTypeParameters: TypeParameter[];      // Local type parameters (undefined if none)
     }
 
     export interface InterfaceTypeWithBaseTypes extends InterfaceType {

```

Comment:
I expected to see things like changes to ClassDeclation/EnumDeclaration/etc. to make them statements.

If they're not statements, then we need to change things like Block to have an array of ModuleElements instead.

Right now the type system isn't correct vis-a-vis what we say you may encounter and what you can actually encounter.